CustomCollections/CustomCollections/CustomHashTable.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+namespace CustomCollections
+{
+    class CustomHashTable<TKey, TValue> : IDictionary
```

Comment:
You need to implement interface IDictionary<TKey,TValue>. If you look into which type it uses for extending ICollection<T> it might also suggest to you which type would be better candidate for internal array instead of Tuple.